packages/web/pages/hustlers/mint-success.tsx

```diff
@@ -188,25 +190,45 @@ const AlertContainer = styled.div`
   animation: appear 0.1s linear 3.1s 1 forwards;
 `;
 
+const ScreenSaver = styled.div<{ image: string }>`
+  position: absolute;
+  top: 0px;
+  left: 0px;
+  width: 100%;
+  height: 100%;
+  z-index: 0;
+  background: ${({ image }) =>
+    `#000000 url('/images/hustler/${image}') center / cover no-repeat fixed;`};
+`;
+
 const MintSuccess = () => {
+  const { account } = useWeb3React();
+  const [hasTransfered, setHasTransfered] = useState(false);
+  const hustler = useHustler();
   const [gangstaParty, setGangstaParty] = useState(false);
 
   const image = gangstaParty ? 'bridge_with_hustlers.png' : 'bridge_no_hustlers.png';
 
-  const ScreenSaver = styled.div`
-    position: absolute;
-    top: 0px;
-    left: 0px;
-    width: 100%;
-    height: 100%;
-    z-index: 0;
-    background: #000000 url('/images/hustler/${image}') center / cover no-repeat fixed;
-  `;
+  const listener = useCallback(
```

Comment:
This approach still subscribes to all events and filters on the client. With the rpc, it is possible to filter at the node level. We should subscribe with a filter scoped to the users address. An example is given here: https://ethereum.stackexchange.com/questions/87643/how-to-listen-to-contract-events-using-ethers-js

Reply:
@smakosh just waiting on the update here before reviewing/merging